spec/helpers/react_helper_spec.rb

```diff
@@ -38,6 +39,56 @@
       end
     end
 
+    context 'when affiliate has alert with text and title' do
+      let(:alert) { Alert.new(text: 'Alert text', title: 'Alert title', affiliate_id: affiliate.id) }
+
+      it 'sets alert to contain both text and title' do
+        helper.search_results_layout(search, {}, vertical, alert.affiliate)
+        expect(helper).to have_received(:react_component).with(
+          'SearchResultsLayout',
+          hash_including(:alert)
```

Comment:
I missed this last time but what do you think about testing that it includes the alert fields as well. e.g.
```ruby
 hash_including(alert: {text: ''Alert text',  title: 'Alert title'})
```

or putting it all together without the `let` statement 

```ruby
  alert_data = {text: ''Alert text',  title: 'Alert title'}
  affiliate.alert.update(alert_data)
  hash_including(alert: alert_data)
```

Same thing for the following test cases